docs/docs/docs/getting_started/writing_first_app/buidling_graph.md

```diff
@@ -0,0 +1,132 @@
+---
+sidebar_position: 2
+title: Building a Graph
+description: Building the Shop Graph with Product and Features.
+---
+
+As the first step of the jac program let's build a graph to the shop.
+
+### Nodes
+
+Nodes can be thought of as the representation of an entity.
+Nodes are the fundamental unit of  a graph. These can be considered to be the steps in which the Walker can take.
+
+* Nodes are composed of Context and executable actions.
+* Nodes execute a set of actions upon entry and exit.
+
+
+The shop graph will have nodes named `product`, `product_catalog`. Let's define them. The has keyword is used to declare a variable for the node. The `product` node has two variables named `name` and `stock`. The default value of `stock` is 0 where the `product_catalog` doesn't have any variables defined yet.
+
+```jac
+node product {
+    has name;
+    has stock = 0;
+}
+
+node product_catalog;
+```
+
+### Edges
+
+Edges are the link between nodes. They walker will use these edges to determine the next node to traverse to.
+The has key word is used to declare the variable "intent". This "intent" is what the Walker will use to to determine which node to go to next.
+
+The shop graph has one type of edges named `category`. We can define that as follows;
+
+```
+edge category {
+    has name;
+}
+```
+
+### Connecting nodes and edges to create a graph
+
+The nodes of are connected to each other with edges to create a graph.
+
+The `graph` is a collection of initialized nodes.
+The `has anchor` key word is used to identify the root node. The Root node is the node where the walker's traversal begins.
+The has anchor key word is used to state the root node. The Root node is the node where the walker's traversal begins.
+
+`spawn` is used to create to create child nodes, which is used to design flow of the conversational experience.
+We are able to create additional edges to connect nodes which which do not share a parent -child relationship.
+
+```jac
+graph shop {
+    has anchor catalog;
+    spawn {
+        catalog = spawn node::product_catalog;
+
+        apple = spawn node::product(name="apple");
+        catalog +[category(name="fruit")]+> apple;
+
+        banana = spawn node::product(name="banana");
+        catalog +[category(name="fruit")]+> banana;
+
+        notebook = spawn node::product(name="notebook");
+        catalog +[category(name="supplies")]+> notebook;
+    }
+}
+```
```

Comment:
Add a bit more explanation of what the code is actually doing, in the context of the application.